keystone/gauss_fit.py

```diff
@@ -0,0 +1,230 @@
+
+	# Find the channel range corresponding to vmin and vmax
+	# -- This is a hold-over from when I originally set up the code to 
+	#    use a channel range rather than velocity range.
+	#    Can change later, but this should work for now. 
+	low_channel = np.where(spectra_x_axis_kms<=vmax)[0][0]+1 # Add ones to change index to channel
+	high_channel = np.where(spectra_x_axis_kms>=vmin)[0][-1]+1 # Again, hold-over from older setup
+	peak_channels = [low_channel, high_channel]
+
+	# Create cubes for storing the fitted Gaussian profiles
+	# and the Gaussians with noise added back into the spectrum
+	header = cube_km.header
+	cube_gauss = np.array(cube_km.unmasked_data[:,:,:])
+	cube_gauss_noise = np.array(cube_km.unmasked_data[:,:,:])
+	shape = np.shape(cube_gauss)
+
+	# Set up a cube for storing fitted parameters
+	param_cube = cube_gauss[0]
+	param_cube = param_cube.reshape((1,) + param_cube.shape)
+	param_cube = np.concatenate((param_cube, param_cube, param_cube, param_cube, param_cube, 	param_cube), axis=0)
+	param_header = cube_km.header
+
+	# Define the Gaussian profile
+	def p_eval(x, a, x0, sigma):
+    		return a*np.exp(-(x-x0)**2/(2*sigma**2))
+
+	# Create some arrays full of NANs
+	# To be used in output cubes if fits fail
+	nan_array=np.empty(shape[0]) # For gauss cubes
+	nan_array[:] = np.NAN
+	nan_array2=np.empty(param_cube.shape[0]) # For param cubes
+	nan_array2[:] = np.NAN
+
+	# Loop through each pixel and find those
+	# with SNR above snr_min
+	x = []
+	y = []
+	pixels = 0
+	for (i,j), value in np.ndenumerate(cube_gauss[0]):
+     		spectra=np.array(cube_km.unmasked_data[:,i,j])
+     		rms = np.std(np.append(spectra[0:(peak_channels[0]-1)], spectra[(peak_channels[1]+1):len(spectra)]))
+     		if (max(spectra[peak_channels[0]:peak_channels[1]]) / rms) > snr_min:
+            		pixels+=1
+	    		x.append(i)
+	    		y.append(j)
+     		else:	
+	    		cube_gauss[:,i,j]=nan_array
+	    		param_cube[:,i,j]=nan_array2
+			cube_gauss_noise[:,i,j]=nan_array
+	print str(pixels) + ' Pixels above SNR=' + str(snr_min) 
+
+	# Define a Gaussian fitting function for each pixel
+	# i, j are the x,y coordinates of the pixel being fit
+	def pix_fit(i,j):
+		spectra = np.array(cube_km.unmasked_data[:,i,j])
+		# Use the peak brightness Temp within specified channel 
+		# range as the initial guess for Gaussian height
+		Tpeak = max(spectra[peak_channels[0]:peak_channels[1]])
+		# Use the velocity of the brightness Temp peak as 
+		# initial guess for Gaussian mean
+		vpeak = spectra_x_axis_kms[peak_channels[0]:peak_channels[1]][np.where(spectra[peak_channels[0]:peak_channels[1]]==Tpeak)]
+		rms = np.std(np.append(spectra[0:(peak_channels[0]-1)], spectra[(peak_channels[1]+1):len(spectra)]))
+		err1 = np.zeros(shape[0])+rms
+		# Create a noise spectrum based on rms of off-line channels
+		# This will be added to best-fit Gaussian to obtain a noisy Gaussian 
+		noise=np.random.uniform(-1.*rms,rms,len(spectra_x_axis_kms))
```

Comment:
I'm unsure as to why you'd want to use uniformly distributed noise here?  Wouldn't you want a normal?

Reply:
Yep, thanks for the catch. I will change this in GAS too